streams/tests/test_dask.py

```diff
@@ -54,6 +55,33 @@ def test_map(c, s, a, b):
 
 
 @gen_cluster(client=True)
+def test_compute(c, s, a, b):
+    source = Stream()
+
+    @delayed
+    def inc1(x):
+        """Immediate delayed function"""
+        return inc(x)
+
+    def inc2(x):
+        """Function that just returns a delayed"""
+        return delayed(inc)(x)
+
+    L1 = source.to_dask().compute(inc1).sink_to_list()
+    L2 = source.to_dask().compute(inc2).sink_to_list()
```

Comment:
Hurt it how?

I agree that this is more complex and that we shouldn't do it if not useful.  I'm thinking about cases where your inputs and outputs come in and leave at different rates.  It looks like your graph has four inputs.  How do you apply this graph onto one element coming through the stream?  It is a tuple of four things maybe?

Also, just to be clear I'm not negative on your PR, just thinking about other alternatives in this space.